### PR TITLE
RP rerun Improvements along with fixes for RP5 API changes 

### PR DESCRIPTION
**Pull request by jyejare**
This has a separate 2 commits for different purposes:

**1. Rerun Failed and Skipped tests simultaniously**
Early, Failed and skipped has to rerun separately in different sessions. But with this PR we should be able to combine both in a same session as :
```
# pytest tests/foreman/ --only-failed --only-skipped
```
Commit : f6f15e2a6dbc8b9ecaa9921b9d8e63aae48db758

**2. RP5 API changes has been fixed in this same PR**

Cause of RP5 changes in APIs, the rerun RP plugin was broken, this PR fixes it.

Commit: c0597b86179cb44df06e85a952584ac31a0c05a1

### CODE REVIEW

File: pytest_plugins/rerun_rp/rerun_rp.py [hunk 1]
```diff
@@ -34,11 +34,13 @@ def _get_tests(launch, **kwargs):
...
     tests = []
     if 'defect_types' in kwargs:
-        defect_types = kwargs.pop('defect_types')
-        for dtype in defect_types:
-            tests.extend(launch.tests(**kwargs, defect_type=dtype).keys())
-    else:
-        tests.extend(launch.tests(**kwargs).keys())
+        for dtype in kwargs['defect_types']:
+            tests.extend(launch.tests(status='failed', defect_type=dtype).keys())
+    if 'status' in kwargs and 'defect_types' not in kwargs:
```
Comment (omkarkhatavkar) [inline pytest_plugins/rerun_rp/rerun_rp.py:39]: `elif 'status' in kwargs:` should work here as well

**Review by omkarkhatavkar — APPROVED** [review]
ACK

**Comment by jyejare** [comment]
@omkarkhatavkar @rplevka @SatelliteQE/airgun-tier-2-reviewers This PR also fixes the RP5 issues.

Here is the latest result on RP5 Report portal instance:

1. with combining Skipped and Failed tests :

```
$ pytest tests/foreman/api/test_discoveredhost.py --only-failed  --only-skipped --collect-only
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.8.3, pytest-6.2.1, py-1.9.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/jyejare/JWorkstation/GitWork/GitRepos/robottelo, configfile: pyproject.toml
plugins: cov-2.10.1, forked-1.3.0, xdist-2.2.0, services-2.2.1, mock-3.4.0, ibutsu-1.13
collected 13 items / 2 deselected / 11 selected                                                                                                                                                                   

<Package api>
  <Module test_discoveredhost.py>
    <Function test_positive_provision_pxe_less_host>
    <Function test_positive_provision_pxe_host[non-admin]>
    <Function test_positive_auto_provision_pxe_host[non-admin]>
    <Function test_positive_reboot_pxe_host[non-admin]>
    <Function test_positive_reboot_all_pxe_hosts[non-admin]>
    <Function test_positive_provision_pxe_host[admin]>
    <Function test_positive_auto_provision_pxe_host[admin]>
    <Function test_positive_reboot_pxe_host[admin]>
    <Function test_positive_reboot_all_pxe_hosts[admin]>
    <Function test_positive_auto_provision_all>
    <Function test_positive_refresh_facts_pxe_host>

============================== 11/13 tests collected (2 deselected) in 159.31s (0:02:39) ====================== 
```
2. Just failed tests :

```

```

3. Just skipped tests:

```

```

**Review by mshriver — APPROVED** [review]

**Comment by jyejare** [comment]
Thanks @mshriver ' y.

**Pull request merged**